source/common/tracing/zipkin/zipkin_tracer_impl.cc

```diff
@@ -12,6 +12,20 @@
 namespace Envoy {
 namespace Zipkin {
 
+namespace {
+// The functions below are needed due to the "C++ static initialization order fiasco."
+
+/**
+ * @return a regex to match a "not sampled" decision.
+ *
+ * Note that a function is needed because the string used to build the regex
+ * cannot be initialized statically.
+ */
+static const std::regex& notSampledRegex() {
```

Comment:
most don't use regex, rather '1' or 'true' literal comparison, anything else false (if present). regex tends to be a lot more expensive and warranted if there's a complex decision. If there's secondary reasons for regex, ignoreme

Reply:
yeah this has already been changed in the main PR